src/olympia/addons/models.py

```diff
@@ -1603,195 +1590,8 @@ class AddonReviewerFlags(ModelBase):
-
-    @cached_property
-    def footer_url(self):
-        return self.footer and self._image_url(self.footer) or ''
-
-    @cached_property
-    def header_path(self):
-        return self._image_path(self.header)
-
-    @cached_property
-    def footer_path(self):
-        return self.footer and self._image_path(self.footer) or ''
-
-    @cached_property
-    def update_url(self):
-        locale = settings.LANGUAGE_URL_MAP.get(trans_real.get_language())
-        url = settings.VAMO_URL + '/%(locale)s/themes/update-check/%(id)d'
-        return url % {
-            'locale': locale or settings.LANGUAGE_CODE,
-            'id': self.addon.id
-        }
-
-    @cached_property
-    def theme_data(self):
-        """Theme JSON Data for Browser/extension preview."""
-        def hexcolor(color):
-            return ('#%s' % color) if color else None
-
-        addon = self.addon
-        return {
-            'id': str(self.addon.id),  # Personas dislikes ints
-            'name': str(addon.name),
-            'accentcolor': hexcolor(self.accentcolor),
-            'textcolor': hexcolor(self.textcolor),
-            'category': (str(addon.all_categories[0].name) if
-                         addon.all_categories else ''),
-            # TODO: Change this to be `addons_users.user.display_name`.
-            'author': self.display_username,
-            'description': (str(addon.description)
-                            if addon.description is not None
-                            else addon.description),
-            'header': self.header_url,
-            'footer': self.footer_url or '',
-            'headerURL': self.header_url,
-            'footerURL': self.footer_url or '',
-            'previewURL': self.preview_url,
-            'iconURL': self.icon_url,
-            'updateURL': self.update_url,
-            'detailURL': jinja_helpers.absolutify(self.addon.get_url_path()),
-            'version': '1.0'
-        }
-
-    @property
-    def json_data(self):
-        """Persona JSON Data for Browser/extension preview."""
-        return json.dumps(self.theme_data,
-                          separators=(',', ':'), cls=AMOJSONEncoder)
-
-    def authors_other_addons(self, app=None):
-        """
-        Return other addons by the author(s) of this addon,
-        optionally takes an app.
-        """
-        qs = (Addon.objects.valid()
-                           .exclude(id=self.addon.id)
-                           .filter(type=amo.ADDON_PERSONA))
-        return (qs.filter(addonuser__listed=True,
-                          authors__in=self.addon.listed_authors)
-                  .distinct())
-
-    @cached_property
-    def listed_authors(self):
-        return self.addon.listed_authors
-
-
 class MigratedLWT(OnChangeMixin, ModelBase):
-    lightweight_theme = models.ForeignKey(
-        Addon, unique=True, related_name='migrated_to_static_theme',
-        on_delete=models.CASCADE)
+    lightweight_theme_id = models.PositiveIntegerField()
```

Comment:
This is so we can keep the MigratedLWT model around and serve updates after the LWT have been hard-deleted.  I'm assuming changing this and dropping the constraint (see migration) is enough.